system-x/services/kudu/src/main/java/software/tnb/kudu/resource/openshift/RemoteClient.java

```diff
@@ -0,0 +1,192 @@
+package software.tnb.kudu.resource.openshift;
+
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.kudu.resource.client.KuduClient;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.awaitility.Awaitility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import io.fabric8.kubernetes.client.dsl.ExecListener;
+import io.fabric8.kubernetes.client.dsl.ExecWatch;
+
+/**
+ * Run commands in the client POD using <a href=https://kudu.apache.org/docs/command_line_tools_reference.html>Kudu CLI</a>
+ */
+public class RemoteClient implements KuduClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteClient.class);
+
+    private final List<String> masters;
+    private final String clientPodName;
+    private final ObjectMapper mapper;
+    private final OpenshiftClient client;
```

Comment:
this is used only once, so maybe it's not worth saving into a variable

Reply:
removed `client`, since it is already checked in `OpenshiftClient`, `mapper` is used 2 times, so I would like to avoid to allocate more than one instance. About `masters` and `clientPodName` those are the mandatory variables, so I prefer to keep them in the constructors as `final`, so that I'm sure I have it in the instance

Reply:
sure, I mean the client only